api/src/routers/meals.js

```diff
@@ -7,66 +7,211 @@ mealsRouter.get("/", async (req, res) => {
   res.json({ Message: "Welcome to meal sharing App" });
 });
 
-mealsRouter.get("/future-meals", async (req, res) => {
+mealsRouter.get("/meals", async (req, res, next) => {
+  const queryParameter = Object.keys(req.query).length > 0;
+  if (!queryParameter) {
+    try {
+      const meals = await knex.from("meal").select("*");
+      res.json(meals);
+    } catch (error) {
+      console.error(error);
+      res.status(404).json({ error: "Request not completed" });
+    }
+  }
+  next();
```

Comment:
Since you are responding to the client above if queryParameter is not present, you should make sure not to call next in that case. So I suggest to wrap next in the else block.

Reply:
But good job on using a middleware and getting practice with it :+1:

Reply:
Thanks for suggestion, It works fine and logically correct.